src/transformers/models/cohere/modeling_cohere.py

```diff
@@ -1165,7 +1166,11 @@ def prepare_inputs_for_generation(
         **kwargs,
     ):
         past_length = 0
-        if past_key_values is not None:
+        # We may have an initialized but empty DynamicCache during first iteration
+        past_exist = past_key_values is not None and not (
+            isinstance(past_key_values, DynamicCache) and len(past_key_values) == 0
+        )
+        if past_exist:
```

Comment:
if we are in generate, this should not even exist anyway and cache positions should be used!

Reply:
this particular piece of code is a blocker for me. It's something we should and can handle in generate no?